src/librustc/ty/maps/plumbing.rs

```diff
@@ -294,24 +306,39 @@ macro_rules! define_maps {
                     )
                 );
 
-                if let Some(value) = tcx.maps.$name.borrow().map.get(&key) {
-                    profq_msg!(tcx, ProfileQueriesMsg::CacheHit);
-                    tcx.dep_graph.read_index(value.index);
-                    return Ok((&value.value).clone());
+                macro_rules! get_lock {
```

Comment:
why is this a macro and not a function?

Reply:
Oh, because it returns I guess? Hmm. OK. Maybe we can rename it `get_lock_or_return` or something.